monkey-runner/plugin/src/main/groovy/com/novoda/monkey/MonkeyConfigurationPlugin.groovy

```diff
@@ -37,60 +36,39 @@ public class MonkeyConfigurationPlugin implements Plugin<Project> {
         def android = project.extensions.findByName("android")
         android.command.devices().eachWithIndex { device, index ->
 
-            def showOverlayTask = project.task("showOverlayDevice${index}", type: NotificationBarOverlay) {
-                show = true
-                deviceId = device.id
-            }
-
             def monkeyTask = project.task("runMonkeyDevice${index}", type: TargetedMonkey) {
                 packageName = extension.packageNameFilter
                 events = extension.eventsCount
                 deviceId = device.id
                 logFileName = extension.logFileName
-                categories = ["android.intent.category.MONKEY"]
-            }
-
-            def hideOverlay = project.task("hideOverlayDevice${index}", type: NotificationBarOverlay) {
-                show = false
-                deviceId = device.id
+                categories = extension.categories
             }
 
             def uninstallApp = project.task("uninstallMonkeyDevice${index}", type: TargetedUninstall) {
                 packageName = extension.packageNameFilter
                 deviceId = device.id
             }
 
-            hideOverlay.dependsOn uninstallApp
-            monkeyTask.dependsOn showOverlayTask
-            monkeyTask.dependsOn extension.taskDependency
-            monkeyTask.finalizedBy hideOverlay
-            runMonkeyAllTask.dependsOn monkeyTask
-        }
-    }
-
-    static class MonkeyRunnerExtension {
-
-        String taskDependency
-        Integer eventsCount
-        String packageNameFilter
-        String logFileName
-
-        void setDefaultsForOptionalProperties() {
-            eventsCount = 50000
-            logFileName = 'monkey.log'
-        }
-
-        void ensureMandatoryPropertiesPresent() {
-            if (taskDependency == null) {
-                notifyMissingProperty('taskDependency')
+            if (extension.useMonkeyTrap) {
+                def showOverlayTask = project.task("showOverlayDevice${index}", type: NotificationBarOverlay) {
+                    show = true
+                    deviceId = device.id
+                }
+
+                def hideOverlay = project.task("hideOverlayDevice${index}", type: NotificationBarOverlay) {
+                    show = false
+                    deviceId = device.id
+                }
+
+                hideOverlay.dependsOn uninstallApp
+                monkeyTask.dependsOn showOverlayTask
+                monkeyTask.finalizedBy hideOverlay
```

Comment:
Well the second branch (`else`) won't have `showOverlay` as we don't want to show it, right ? So it won't make sense both branches to depend on `hideOverlay` since we don't show it.